docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+    import "../token.sol" as token; // source unit name: token.sol
+
+.. note::
+
+    Do not confuse relative imports with relative paths.
+    Both ``util.sol`` and ``./util.sol`` specify relative paths on disk but these paths are treated
+    very differently when used in imports.
+    Only the latter creates a relative import.
+
+    Consider the following example:
+
+    .. code-block:: solidity
+       :caption: /project/lib/math.sol
+
+       import "/project/lib/util.sol" as util1; // source unit name: /project/lib/util.sol
+       import "./util.sol" as util2;            // source unit name: /project/lib/util.sol
+       import "util.sol" as util3;              // source unit name: util.sol
+
+    In the situation above the first and the second import are equivalent and refer to the same
+    source unit in the virtual filesystem.
+    The compiler will recognize that the source has already been compiled when it encounters
+    ``./util.sol`` and will not try to compile it again.
+    This is not the case with the third import.
+    When asked for ``util.sol`` with a direct import, the compiler will try to find exactly that.
+    The entry with the source unit name of ``/project/lib/util.sol`` will not be used.
+
+    Even if you run the compiler from within ``/project/lib/`` the relative ``util.sol`` will only
+    get resolved into ``/project/lib/util.sol`` by the file loader.
+    When the loader returns the source, the compiler will compile it but still place it under
+    ``util.sol`` and not ``/project/lib/util.sol`` in the virtual filesystem.
+
+Unlike in direct imports, the paths used in relative imports do get normalized.
+The normalization rules are the same as for UNIX paths, namely:
+
+- All the ``./`` segments are removed.
+- Every ``../``  segment backtracks one level up in the hierarchy.
+- Multiple slashes are squashed into a single one.
+
+Example:
+
+.. code-block:: solidity
+    :caption: lib/contract.sol
+
+    import "./util/./util.sol";         // source unit name: lib/util/util.sol
+    import "./util//util.sol";          // source unit name: lib/util/util.sol
+    import "../util/../array/util.sol"; // source unit name: array/util.sol
+
+.. warning::
+
+    The root of the virtual filesystem is an empty path, not ``/``.
+    This matters when the ``../`` segments go beyond the root.
+    In UNIX paths such segments are ignored and for example ``/../../`` is
+    equivalent to just ``/``.
+    In the virtual filesystem the rule similar but the result is an empty path instead.
+
+    .. code-block:: solidity
+        :caption: /project/lib/contract.sol
+
+        import "../util.sol";          // source unit name: /project/util.sol
+        import "../../util.sol";       // source unit name: /util.sol
+        import "../../../util.sol";    // source unit name: util.sol
+        import "../../../../util.sol"; // source unit name: util.sol
+
+.. note::
+
+    The parent source unit name is **not** normalized.
```

Comment:
"parent source unit name" is not properly defined. Maybe just use "importing source unit name"?

Reply:
Done.